client/src/App.js

```diff
@@ -1,15 +1,51 @@
 import { Routes, Route } from 'react-router-dom';
 import Home from './pages/home';
 import Product from './pages/product';
+import Cart from './pages/cart';
 import styled from 'styled-components';
 import { tastes } from './database.js';
+import { useState } from 'react';
+
 
 function App() {
+  
+  const [cartItems, setCartItems] = useState([]);
+
+  const onAdd = tasteSample => {
+    const exist = cartItems.find(item => item.id === tasteSample.id);
+    if (exist) {
+      setCartItems(
+        cartItems.map(item => (item.id === exist.id ? { ...exist, qty: exist.qty + 1 } : item))
+      );
+    } else {
+      setCartItems([...cartItems, { ...tasteSample, qty: 1 }]);
+    }
+  };
+
+  const onRemove = tasteSample => {
+    const exist = cartItems.find(item => item.id === tasteSample.id);
+    if (exist.qty > 1) {
+      setCartItems(
+        cartItems.map(item => (item.id === exist.id ? { ...exist, qty: exist.qty - 1 } : item))
+      );
+    } else {}
+  };
+
+  const onDelete = tasteSample => {
+    const exist = cartItems.find(item => item.id === tasteSample.id);
```

Comment:
Same here, I think "exist" is not a good naming choice.

Reply:
Thank you Paul, that's a good point. I changed all "exist" consts into "existingItem"